faith_presentation/src/commonMain/kotlin/net/thechance/mena/faith/presentation/feature/prayertime/PrayerTimeViewModel.kt

```diff
@@ -100,6 +105,15 @@ class PrayerTimeViewModel(
         )
     }
 
+    private fun getNextDayPrayerTimesIfNoMoreInCurrent(prayerTime: PrayerTime) {
+        val currentDate = uiState.value.currentDate
+        val prayerTimeDay = getHijriDay(prayerTime)
+        if (currentDate.day >= prayerTimeDay) return
+        val nextIslamicDate = currentDate.copy(day = currentDate.day.inc())
+        updateState { it.copy(currentDate = nextIslamicDate) }
+        getPrayerTimesForIslamicDate(nextIslamicDate)
+    }
+
```

Comment:
this fun have a Hidden Side Effects 
` updateState { it.copy(currentDate = nextIslamicDate) }`


```suggestion
       private fun advanceToNextDayIfNeeded(lastPrayerTime: PrayerTime) {
    if (shouldAdvanceToNextDay(lastPrayerTime)) {
        advanceToNextDay()
    }
}

private fun shouldAdvanceToNextDay(lastPrayerTime: PrayerTime): Boolean {
    val currentDay = uiState.value.currentDate.day
    val lastPrayerDay = getHijriDay(lastPrayerTime)
    return currentDay < lastPrayerDay
}

private fun advanceToNextDay() {
    val nextDate = uiState.value.currentDate.incrementDay()
    updateState { it.copy(currentDate = nextDate) }
    getPrayerTimesForIslamicDate(nextDate)
}
```
can you use this refactord fun and read [this](https://thechance-workspace.slack.com/archives/C08RJB00449/p1763621145448139)

Reply:
check now